composer/trainer/trainer.py

```diff
@@ -278,6 +279,8 @@ def _get_ddp_sync_strategy(ddp_sync_strategy: Optional[Union[str, DDPSyncStrateg
 
 
 def _generate_run_name() -> str:
+    # change coolname randomness for different names with same seed
+    coolname.replace_random(random.Random(os.urandom(128)))
     # prefixing with the time so experiments sorted alphabetically will have the latest experiment last
     generated_run_name = str(int(time.time())) + '-' + coolname.generate_slug(2)
     run_name_list = [generated_run_name]
```

Comment:
@hanlint  did you get those run_names from running it? I believe the lines below here (github wont let me highlight for some reason) should take `run_name` generated on rank0 and override the `run_name`s generated by other ranks

Reply:
Ah, no I took the opportunity to make those up :) If it's already handled/tested with DDP, then great

Reply:
LOL

Reply:
I was very unsure because the names were solid

Reply:
I'm re-enabling automerge and requesting review!